packages/welcome-screen/src/App.vue

```diff
@@ -11,78 +11,46 @@
 
 export default {
   name: 'App',
-  data() {
-    return {
-      firstLaunch: false,
-    };
-  },
   created () {
     document.title = "Welcome";
   },
   mounted() {
-    window.show_retry = () => {
-      this.$router.push('/loading/retry');
-    }
-
-    window.show_error = () => {
-      this.$router.push('/loading/error');
-    }
-
-    window.show_loading = () => {
-      this.$router.push('/loading/default');
-    }
-
-    window.clean = () => {
-      this.$router.push('/loading/default');
-    }
-
-    window.show_welcome = () => {
-      this.$router.push('/welcome');
-    }
-
-    window.show_endless_key = () => {
-      this.$router.push('/endless-key');
-    }
-
-    window.show_endless_key_required = () => {
-      this.$router.push('/endless-key-required');
-    }
-
-    window.show_permissions_cancelled = () => {
-      this.$router.push('/permissions-cancelled');
-    }
-
-    window.show_wrong_folder = () => {
-      this.$router.push('/permissions-wrong-folder');
-    }
-
-    window.setHasUSB = (isUsbConnected) => {
-      this.$store.commit('setUsbConnected', isUsbConnected);
-    }
-
-    window.setNeedsPermission = (needsPermission) => {
-      this.$store.commit('setNeedsPermission', needsPermission);
-    }
-
-    window.firstLaunch = () => {
-      this.firstLaunch = true;
-      this.$router.push('/welcome');
-    }
-  },
-  beforeDestroy() {
-    function fallback() {
-      console.log('No welcome screen');
+    window.WelcomeApp = {
+      showLoadingRetry: () => {
```

Comment:
Did you explore using [provide / inject](https://vuejs.org/guide/components/provide-inject.html) here, instead of putting an object in `window`? I'm curious if that would work in a more Vue-ish way :)

Reply:
We are putting an object in `window` because it needs to be called from the Android/Windows apps.

From Android:
```
mWebView.evaluateJavascript("WelcomeApp.showWelcome()", null);
```

From Windows:
```
mainWindow.webContents.executeJavaScript('WelcomeApp.showWelcome()', true);
```

(See PRs https://github.com/endlessm/kolibri-installer-android/pull/98 and https://github.com/endlessm/endless-key-app/pull/96 which should be reviewed along this one as mentioned in the ticket https://phabricator.endlessm.com/T34380#966384)

I went through the provide/inject docs and I didn't see how it can replace this.